persist/fs/commitlog/commit_log.go

```diff
@@ -187,19 +197,18 @@ func (l *commitLog) flushEvery(interval time.Duration) {
 		lastFlushAt := l.lastFlushAt
 		l.flushMutex.RUnlock()
 
-		sinceLastFlush := l.nowFn().Sub(lastFlushAt)
-		if sinceLastFlush < interval {
+		if sinceFlush := l.nowFn().Sub(lastFlushAt); sinceFlush < interval {
 			// Flushed already recently, sleep until we would next consider flushing
-			sleepForOverride = interval - sinceLastFlush
+			sleepForOverride = interval - sinceFlush
 			continue
 		}
 
 		// Request a flush
-		l.RLock()
-		if l.closed {
+		if l.RLock(); l.closed {
```

Comment:
Any reason to combine these?  Seems like `l.RLock()` is not really a part of the condition.